cr_module/classes/redfish.py

```diff
@@ -581,6 +581,16 @@ def determine_vendor(self):
 
             self.vendor_dict_key = vendor_string
 
+        if vendor_string == "" and self.connection.root.get("Chassis") is not None:
+            chassis = self.get_system_properties("chassis")
+            if len(chassis) > 0:
+                # try to get vendor from first chassis
+                chassis_data = self.get(chassis[0])
+                if chassis_data is not None:
+                    vendor_string = chassis_data.get("Manufacturer", None)
```

Comment:
this could probably be compacted to:
```python
vendor_string = grab(self.get_system_properties, "chassis.0.Manufacturer")
```

Reply:
The problem here is that we don’t have the chassis data at this point. We have to fetch it from Redfish first.
So I have updated the code to be more concise using `grab`, but still we have to call the GET first. Or am I missing something?